packages/rum-core/src/boot/preStartRum.ts

```diff
@@ -0,0 +1,169 @@
+
+    bufferApiCalls.drain(startRumResult)
+  }
+
+  return {
+    init(initConfiguration) {
+      if (!initConfiguration) {
+        display.error('Missing configuration')
+        return
+      }
+
+      const eventBridgeAvailable = canUseEventBridge()
+      if (eventBridgeAvailable) {
+        initConfiguration = overrideInitConfigurationForBridge(initConfiguration)
+      }
+
+      // Expose the initial configuration regardless of initialization success.
+      cachedInitConfiguration = initConfiguration
+
+      if (cachedConfiguration) {
+        displayAlreadyInitializedError('DD_RUM', initConfiguration)
+        return
+      }
+
+      // If we are in a Synthetics test configured to automatically inject a RUM instance, we want to
+      // completely discard the customer application RUM instance by ignoring their init() call.  But,
+      // we should not ignore the init() call from the Synthetics-injected RUM instance, so the
+      // internal `ignoreInitIfSyntheticsWillInjectRum` option is here to bypass this condition.
+      if (ignoreInitIfSyntheticsWillInjectRum && willSyntheticsInjectRum()) {
+        return
+      }
+
+      const configuration = validateAndBuildRumConfiguration(initConfiguration)
+      if (!configuration) {
+        return
+      }
+
+      if (!eventBridgeAvailable && !configuration.sessionStoreStrategyType) {
+        display.warn('No storage available for session. We will not send any data.')
+        return
+      }
+
+      if (configuration.compressIntakeRequests && !eventBridgeAvailable && startDeflateWorker) {
+        deflateWorker = startDeflateWorker(
+          configuration,
+          'Datadog RUM',
+          // Worker initialization can fail asynchronously, especially in Firefox where even CSP
+          // issues are reported asynchronously. For now, the SDK will continue its execution even if
+          // data won't be sent to Datadog. We could improve this behavior in the future.
+          noop
+        )
+        if (!deflateWorker) {
+          // `startDeflateWorker` should have logged an error message explaining the issue
+          return
+        }
+      }
+
+      cachedConfiguration = configuration
+      tryStartRum()
+    },
+
+    get initConfiguration() {
+      return cachedInitConfiguration
+    },
+
+    getInternalContext: noop as () => undefined,
+
+    stopSession: noop,
+
+    addTiming(name, time = timeStampNow()) {
+      bufferApiCalls.add((startRumResult) => startRumResult.addTiming(name, time))
+    },
+
+    startView(options, startClocks = clocksNow()) {
+      let ignore = false
+      if (!firstStartView) {
+        firstStartView = {
+          options,
+          ignore() {
+            ignore = true
+          },
+        }
+        tryStartRum()
+      }
+
+      bufferApiCalls.add((startRumResult) => {
+        if (!ignore) {
+          startRumResult.startView(options, startClocks)
+        }
+      })
+    },
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Maybe we could direcly use the `firstStartView` ref instead of the `ignore` flag and remove the ref from the bufferApiCalls in case of trackViewsManually is used. Something like:

```typescript
startView(options, startClocks = clocksNow()) {
      const bufferedStartView = (startRumResult: StartRumResult) => {
        startRumResult.startView(options, startClocks)
      }
      if (!firstStartView) {
        firstStartViewOption = options
        firstStartView = bufferedStartView
      }

      bufferApiCalls.add(bufferedStartView)
    }
```

and in `tryStartRum`

```typescript
 if (cachedConfiguration.trackViewsManually) {
      bufferApiCalls.remove(firstStartView!)
      initialViewOptions = firstStartViewOption
    }
```